libraries/ESP8266WebServer/src/detail/RequestHandlersImpl.h

```diff
@@ -15,16 +15,55 @@ class FunctionRequestHandler : public RequestHandler {
     , _uri(uri)
     , _method(method)
     {
+		int numParams = 0, start = 0;
+        do {
+            start = _uri.indexOf("{}", start);
+            if (start > 0) {
+                numParams++;
+                start += 2;
+            }
+        } while (start > 0);
+        pathArgs = std::vector<String>(numParams);
     }
 
     bool canHandle(HTTPMethod requestMethod, String requestUri) override  {
         if (_method != HTTP_ANY && _method != requestMethod)
             return false;
 
-        if (requestUri != _uri)
-            return false;
+        if (_uri == requestUri)
+            return true;
+
+        size_t uriLength = _uri.length();
+        unsigned int pathArgIndex = 0;
+        unsigned int j = 0;
```

Comment:
could `j` be renamed to something more explicit ?
some more comments are welcome

Reply:
renamed j to requestUriIndex